manifests/params.pp

```diff
@@ -158,6 +158,13 @@
   $dhcp_interface        = pick(fact('networking.primary'), 'eth0')
   $dhcp_option_domain    = [$facts['networking']['domain']]
   $dhcp_failover_address = fact('ipaddress')
+  if $foreman_proxy::template and $foreman_proxy::dhcp_ipxe_bootstrap {
+    $dhcp_ipxefilename = "${template_url}/unattended/iPXE?bootstrap=1"
+  } elsif $foreman_proxy::template and !$foreman_proxy::dhcp_ipxe_bootstrap {
+    $dhcp_ipxefilename = "${template_url}/unattended/iPXE"
+  } else {
+    $dhcp_ipxefilename = undef
+  }
```

Comment:
You can't do this logic in `params.pp`. At this point in the evaluation these parameters are unavailable. I would expect this logic to be in `manifests/proxydhcp.pp`.